nl_server/tests/merge_test.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for EmbeddingsResult merging."""
+
+import unittest
+
+from parameterized import parameterized
+
+from nl_server.embeddings import EmbeddingsMatch
+from nl_server.merge import merge_search_results
+
+
+class TestMerge(unittest.TestCase):
+
+  @parameterized.expand([[
+      [{
+          'not poor': [
+              EmbeddingsMatch(sentence='poverty',
+                              score=0.85,
+                              vars=['WHO_Poverty']),
+              EmbeddingsMatch(sentence='above poverty',
+                              score=0.7,
+                              vars=['WHO_AbovePoverty']),
+          ]
+      }, {
+          'not poor': [
+              EmbeddingsMatch(sentence='below poverty',
```

Comment:
This is an edge case but do we want to address a scenario where the same SV is in both inputs?

i.e. the same embeddings match (say `EmbeddingsMatch('below poverty', 0.9, ['Count_BelowPovertyLevel'])` exists in both? Currently the output list will return this match twice. Do we want the merge function to account for duplicates?

Reply:
There are a few edge cases:
1.  Same sentence appears in custom DC and base DC with different SVs
2. Same variable appears in custom DC and base DC with different sentences
3. Same sentence + variable appear

`_rank_vars` will group by SVs and rerank them, and should handle (1) + (2).  (3) is a good point, I've handled that.